gait3d/gaitanalysis_with_top_cam.py

```diff
@@ -0,0 +1,996 @@
+            for key, value in normalization_params.items():
+                additional_logs_to_add[key] = value
+            for key, value in normalized_maze_corner_coords.items():
+                additional_logs_to_add[f'normalized_{key}_coords'] = value
+            self._add_to_logs(logs_to_add = additional_logs_to_add)
+
+
+    def _add_to_logs(self, logs_to_add: Dict) -> None:
+        if hasattr(self, 'logs') == False:
+            self.logs = {}
+        for key, value in logs_to_add.items():
+            assert key not in self.logs.keys(), f'{key} is already in self.logs.keys - adding it would result in overwriting the previous entry; please ensure unique naming'
+            self.logs[key] = value           
+            
+            
+    def _get_max_odd_n_frames_for_time_interval(self, fps: int, time_interval: 0.5) -> int:
+        assert type(fps) == int, '"fps" has to be an integer!'
+        frames_per_time_interval = fps * time_interval
+        if frames_per_time_interval % 2 == 0:
+            max_odd_frame_count = frames_per_time_interval - 1
+        elif frames_per_time_interval == int(frames_per_time_interval):
+            max_odd_frame_count = frames_per_time_interval
+        else:
+            frames_per_time_interval = int(frames_per_time_interval)
+            if frames_per_time_interval % 2 == 0:
+                max_odd_frame_count = frames_per_time_interval - 1
+            else:
+                max_odd_frame_count = frames_per_time_interval
+        assert max_odd_frame_count > 0, f'The specified time interval is too short to fit an odd number of frames'
+        return int(max_odd_frame_count)                
+
+
+    def _get_preprocessing_relevant_marker_ids(self, df: pd.DataFrame) -> List[str]:
+        all_marker_ids = self._get_all_unique_marker_ids(df = df)
+        relevant_marker_ids = all_marker_ids
+        for marker_id_to_exclude in self.marker_ids_to_exclude_for_smoothing_and_interpolation:
+            relevant_marker_ids.remove(marker_id_to_exclude)
+        return relevant_marker_ids
+
+
+    def _get_all_unique_marker_ids(self, df: pd.DataFrame) -> List[str]:
+        unique_marker_ids = []
+        for column_name in df.columns:
+            marker_id, _ = column_name.split('_')
+            if marker_id not in unique_marker_ids:
+                unique_marker_ids.append(marker_id)
+        return unique_marker_ids
+
+
+    def _smooth_tracked_coords_and_likelihood(self, marker_ids: List[str], window_length: int, polyorder: int=3) -> pd.DataFrame:
+        """
+        Smoothes the DataFrame basically using the implementation from DLC2kinematics:
+        https://github.com/AdaptiveMotorControlLab/DLC2Kinematics/blob/82e7e60e00e0efb3c51e024c05a5640c91032026/src/dlc2kinematics/preprocess.py#L64
+        However, with one key change: likelihoods will also be smoothed.
+        In addition, we will not smooth the columns for the tracked LEDs and the MazeCorners.
```

Comment:
Since the maze corners should be stationary I thought it would only decrease the detection performance if we increase points between two stationary detections.